python/ray/serve/_private/proxy.py

```diff
@@ -323,110 +328,94 @@ def _ongoing_requests_end(self):
-                        deployment_name="",
-                        route=proxy_request.route_path,
-                    ),
-                    should_record_access_log=True,
-                    should_record_request_metrics=True,
-                    should_increment_ongoing_requests=False,
-                )
-            else:
-                route_prefix, handle, app_is_cross_language = matched_route
-                # Modify the path and root path so that reverse lookups and redirection
-                # work as expected. We do this here instead of in replicas so it can be
-                # changed without restarting the replicas.
-                route_path = proxy_request.route_path
-                if route_prefix != "/" and self.protocol == RequestProtocol.HTTP:
-                    assert not route_prefix.endswith("/")
-                    proxy_request.set_path(route_path.replace(route_prefix, "", 1))
-                    proxy_request.set_root_path(proxy_request.root_path + route_prefix)
-
-                handle, request_id = self.setup_request_context_and_handle(
-                    app_name=handle.deployment_id.app,
-                    handle=handle,
-                    route_path=route_path,
-                    proxy_request=proxy_request,
-                )
+        if proxy_request.is_health_request:
+            response_generator = self.health_response(healthy=healthy, message=message)
+        else:
+            response_generator = self.routes_response(healthy=healthy, message=message)
 
-                response_generator = self.send_request_to_replica(
-                    request_id=request_id,
-                    handle=handle,
-                    proxy_request=proxy_request,
-                    app_is_cross_language=app_is_cross_language,
-                )
+        return ResponseHandlerInfo(
+            response_generator=response_generator,
+            metadata=HandlerMetadata(
+                route=proxy_request.route_path,
+            ),
+            should_record_access_log=False,
+            should_record_request_metrics=True,
+            should_increment_ongoing_requests=False,
+        )
 
-                return ResponseHandlerInfo(
-                    response_generator=response_generator,
-                    metadata=HandlerMetadata(
-                        application_name=handle.deployment_id.app,
-                        deployment_name=handle.deployment_id.name,
-                        route=route_path,
-                    ),
-                    should_record_access_log=True,
-                    should_record_request_metrics=True,
-                    should_increment_ongoing_requests=True,
-                )
+    def _get_response_handler_info(
+        self, proxy_request: ProxyRequest
+    ) -> ResponseHandlerInfo:
+        if proxy_request.is_health_request or proxy_request.is_route_request:
+            return self._get_health_or_routes_reponse(proxy_request)
+
```

Comment:
The code after this point is unchanged, it was just un-indented due to pulling the above into a separate method